lib/segmentio/index.js

```diff
@@ -162,6 +162,7 @@ Segment.prototype.normalize = function(msg){
   msg.userId = msg.userId || user.id();
   msg.anonymousId = user.anonymousId();
   msg.messageId = uuid();
+  msg.sentAt = new Date();
```

Comment:
can you elaborate more on network failure?
i think for most libs if there's a network failure the event is lost, unless i'm misunderstanding you haha